ws-server/src/server.rs

```diff
@@ -87,6 +92,85 @@ impl SubscriptionSink {
 
 		Ok(())
 	}
+
+	/// Extract subscriptions with input.
+	/// Usually, you want process the input before sending back
+	/// data on that subscription.
+	pub fn extract_with_input(&self) -> Vec<InnerSinkWithParams<P>> {
```

Comment:
Still struggling to grok the impact of this. It looks like it could be a pretty big performance hit to loop over all the subscriptions to find the ones using parameters?

Reply:
That's a good comment, I was mostly thinking that the `Mutex` is the bottleneck in this context.

But again, this code is very "draftisch" and I haven't really put in much effort into mostly to show a point.
I guess we could put `with params` and `params` in different containers to avoid `O(n)` lookups or something

Reply:
Best way is ofc to benchmark this :)